src/geometry/partitions/distributedptscotchunstructuredpartition.h

```diff
@@ -0,0 +1,291 @@
+        {
+            verttabGra[i] = currentEdge;
+
+            for (int other : this->adjacency[start + i])
+            {
+                edgetabGra[currentEdge++] = other;
+            }
+        }
+
+        numEdges = currentEdge;
+
+        verttabGra[localCells] = currentEdge;
+
+        int ierr = 0;
+        error = SCOTCH_dgraphBuild(&graph,
+                0,              // c++ starts counting at 0
+                localCells,     // number of local vertices
+                localCells,     // number of maximum local vertices to be created
+                verttabGra,     // vertex data
+                nullptr,        // these are some ghost zone
+                nullptr,        // and other hints that are optional, so -> nullptr
+                nullptr,        //
+                numEdges,       // number of local edges
+                numEdges,       // number of maximum local edges to be created
+                edgetabGra,     // edge data
+                nullptr,        // more optional data/hints
+                nullptr);       //
+        if (error) std::cout << "SCOTCH_graphBuild error: " << error << ", ierr: " << ierr << std::endl;
+
+        error = SCOTCH_dgraphCheck(&graph);
+        if (error) std::cout << "SCOTCH_graphCheck error: " << error << std::endl;
+
+        SCOTCH_Strat *straptr = SCOTCH_stratAlloc();
+        error = SCOTCH_stratInit(straptr);
+        if (error) std::cout << "SCOTCH_stratInit error: " << error << std::endl;
+
+        error = SCOTCH_dgraphPart(&graph, numPartitions, straptr, &indices[0]);
+        if (error) std::cout << "SCOTCH_graphMap error: " << error << std::endl;
+
+        SCOTCH_dgraphExit(&graph);
+        SCOTCH_stratExit(straptr);
+        delete[] verttabGra;
+        delete[] edgetabGra;
+    }
+
+    void createRegions(const std::vector<SCOTCH_Num> &indices)
+    {
+        // regions won't be complete after partitioning, fragments of different regions
+        // will end up on different nodes and they have to be synchronized.
+        // build partial regions
+        std::vector<Region<1>> partials(numPartitions);
+        for (int i = 0; i < indices.size(); ++i)
+        {
+            partials.at(indices[i]) << Coord<1>(start + i);
+        }
+        regions = partials;
+
+        // send partial regions to all other nodes so they can build the complete regions
+        size_t numIndices = indices.size();
+        for (size_t j = 0; j < numPartitions; ++j)
+        {
+            if (j != MPILayer().rank()) // dont send own regions to self
+            {
+                // send all partial regions
+                for (size_t k = 0; k < numPartitions; ++k)
+                {
+                    MPILayer().sendRegion(partials.at(k), j);
+                }
+            }
+            else
+            {
+                // regions will be received from all ranks except self
+                for (int i = 0; i < numPartitions; ++i)
```

Comment:
This is essentially an MPI_Alltoallv() implemented via P2P communication. Could you file a ticket and assign it to me that the MPILayer should support such an operation for Regions natively?